rmw_dds_common/test/test_graph_cache.cpp

```diff
@@ -0,0 +1,1113 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstring>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "rmw/qos_profiles.h"
+#include "rmw/topic_endpoint_info.h"
+#include "rmw/topic_endpoint_info_array.h"
+
+#include "rmw_dds_common/graph_cache.hpp"
+
+using rmw_dds_common::GraphCache;
+
+struct NameAndNamespace
+{
+  std::string namespace_;
+  std::string name;
+};
+
+MATCHER_P2(IsNameAndNamespace, namespace_, name, "")
+{
+  return (arg.namespace_ == namespace_) && (arg.name == name);
+}
+
+using NamesAndNamespaces = std::vector<NameAndNamespace>;
```

Comment:
@ivanpauno nit: this type alias (and others below) are literally one Hamming distance away from the non-`std::vector` one. Consider just using `std::vector<T>` in full.